packages/core/src/notifications/panelNode.ts

```diff
@@ -207,7 +210,10 @@ export class NotificationsNode implements TreeNode {
                                     )
                                     break
                                 case 'updateAndReload':
-                                    await this.updateAndReload(notification.displayIf.extensionId)
+                                    // Give things time to finish executing.
+                                    globals.clock.setTimeout(() => {
+                                        void this.updateAndReload(notification.displayIf.extensionId)
+                                    }, 1000)
```

Comment:
is there state that can be checked via `waitUntil` ?

Reply:
Some arbitrary amount of telemetry, e.g. a ui_click and the run block it is currently in. Maybe we could check this telemetry state. A tradeoff between complexity <> and potential race conditions.